pkg/cluster/cluster.go

```diff
@@ -148,6 +151,10 @@ func (c *Cluster) create() error {
 		return fmt.Errorf("failed to update cluster phase (%v): %v", spec.ClusterPhaseCreating, err)
 	}
 
+	if err := c.gc.CollectCluster(c.cluster.Name, c.cluster.UID); err != nil {
+		return fmt.Errorf("failed to garbage collect resources: %v", err)
```

Comment:
"clean up conflict resources" is confusing. Let's use "garbage collect resources".